### PR TITLE
Improve the user error message if the upload-dar command fails.

### PR DESCRIPTION
**Pull request by nickchapman-da**
In particular we want to catch the case of a gRPC timeout exception which we know can occur if the DAR is too large. But it seems helpful to catch any exception, and shows what we caught along with the helpful message.

This PR addresses issue: https://github.com/digital-asset/daml/issues/6979

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs [hunk 1]
```diff
@@ -125,7 +136,7 @@ runLedgerUploadDar flags darPathM = do
         getDarPath
     putStrLn $ "Uploading " <> darPath <> " to " <> show hp
     bytes <- BS.readFile darPath
-    uploadDarFile hp bytes
+    uploadDarFile hp bytes `catch` \e -> throw (ExceptionDuringUploadDar e)
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs:139]: Please use `catch` from `safe-exceptions` or even better `catchIO`. You don’t want to catch asynchronous exceptions here.

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs [hunk 2]
```diff
@@ -14,6 +14,7 @@ module DA.Daml.Helper.Ledger (
     runLedgerNavigator
     ) where
 
+import Control.Exception (SomeException,Exception,catch,throw)
```
Comment (sofiafaro-da) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs:17]:
```suggestion
import Control.Exception.Safe (SomeException,Exception,catch,throw)
```

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs (hunk 1 above)
Comment (nickchapman-da) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs:139]: I switch to safe-exceptions.
I couldn't use `catchIO` as this does not catch `GRPCIOTimeout` which is the one I really care about.

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs (hunk 2 above)
Comment (nickchapman-da) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs]: did this

**Review by cocreature — APPROVED** [review]
Thanks!

**Review by hurryabit — COMMENTED** [review]
Nice. Merci.

**Pull request merged**